certbot/eff.py

```diff
@@ -71,11 +71,17 @@ def _check_response(response):
 
     """
     logger.debug('Received response:\n%s', response.content)
-    if response.ok:
-        if not response.json()['status']:
-            _report_failure('your e-mail address appears to be invalid')
-    else:
+    try:
+        response.raise_for_status()
+        assert response.json()['status'] == True
```

Comment:
This is really a nitpick, but you can actually disable [assert statements](https://docs.python.org/2/reference/simple_stmts.html#the-assert-statement) with a command line flag to `python` so I don't think we should rely on them for production behavior.

I'd personally fix this by not raising an exception when `status` is `False` and just use the value in an `if` statement to call `_report_failure`.